packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
+    id: string;
+    type: any;
+    controller?: any;
+    privateKey?: string;
+    privateKeyBase58?: string;
```

Comment:
the word "node" comes form the idea that there is an information model of key pairs as linked data graphs.

the folks at digital bazaar have historically been extremely hostile for thinking about private / secret keys and linked data...

We created the universal wallet spec in part because of how much we disagree on that point, but its worth noting there.